### PR TITLE
stop_dplyr() experiment

### PR DESCRIPTION
**Pull request by romainfrancois**
Writing `cnd_header()` and `cnd_body()` seems to be a lot of work. Unless we want to be able to specifically catch typed dplyr error this might be enough : 

 - the first parameter is the error class, then always appended to "dplyr_error"
 - then a set of glue strings, the data taken from the parent environment, so that we can write small error functions, e.g. 

```r
stop_filter_named <- function(index, expr, name) {
  stop_dplyr("dplyr_error_filter_named",
    "`filter()` argument `..{index}` is named",
    i = "This usually means that you've used `=` instead of `==`",
    i = "Did you mean `{name} == {as_label(expr)}` ?"
  )
}
```

or should functions like `stop_filter_named()` directly call `abort` and include individual `glue()/glue_data()` calls ?

Would like to get this right before starting to rewrite error messages with bullets

### CODE REVIEW

**Comment by hadley** [comment]
I think that's fine, with the small caveat that if we end up using the same error message in multiple places we'll want to wrap it up into a function (but that's only true for relatively few functions).

File: R/conditions.R [hunk 1]
```diff
@@ -1,112 +1,43 @@
...
-cnd_body.dplyr_error_filter_size <- function(cnd) {
-  bullets <- c(
-    x = glue_data(cnd, "It must be of size {expected_size} or 1, not size {size}")
-  )
-  if (is_grouped_df(cnd$data)) {
-    bullets <- c(bullets,
-      i = glue_data(cnd, "The error occured in group {index_group}")
-    )
-  }
-  format_error_bullets(bullets)
```
Comment (hadley) [inline R/conditions.R]: How come we don't need `format_error_bullets()` anywhere anyy more?

Comment (romainfrancois) [reply R/conditions.R]: `rlang::abort()` does it : 

``` r
rlang::abort(c("lorem ipsum", x = "dolor sit amet, consectetur adipiscing elit.", i = "Sed non risus. Suspendisse lectus tortor"))
#> Error: lorem ipsum
#> x dolor sit amet, consectetur adipiscing elit.
#> ℹ Sed non risus. Suspendisse lectus tortor
```

<sup>Created on 2020-01-09 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0.9000)</sup>

**Comment by romainfrancois** [comment]
Factored out `glue_c()` and then just calling `abort()`, feels better than introducing `stop_dplyr()` for no real reason and potentially we can also use `glue_c()` for `warn()` ...

**Comment by romainfrancois** [comment]
Merging this now so I can tackle error messages for other functions. Might be that we just revert to use `abort(c(glue())` though instead of introducing `glue_c()`

**Pull request merged**